### PR TITLE
[EM-902] Add alipayplus wallets for SG

### PR DESCRIPTION
**Pull request by kan98**
#### 1. Objective

Explain in non-technical terms **WHY this PR is required**.
E.g.: What feature it adds, what problem it solves...

This section will be used in the release notes. 

Added support for Alipay, AlipayHK, DANA, GCash, Kakao Pay, TNG eWallet payment methods for Singapore PSP.

**Related information**:
Related issue(s): #< GitHub ticket number > (optional)
https://omise.atlassian.net/browse/EM-902

#### 2. Description of change

A general description of **WHAT changed in the codebase**, but short of an English version of the diff. Assume that people reading this will also be looking at the output of `git diff` and guide them to the highlights.

Additionally add the reasoning for change details if they're complex or abstract.

Added new payment methods to the plugin.

#### 3. Quality assurance

Specify where and how you tested this and what further testing it might need.

Tested on my local instance the different payment methods.

**🔧 Environments:**

Specify the details of your test environments, including, for each, the platform version (on which the plugin was run), the Omise plugin version, and the versions of your system software such as PHP or Ruby.

i.e.
- **Platform version**: Magento CE 2.2.2
- **PHP version**: 7.1

**✏️ Details:**

Explain how to manually test this feature.
For example if changes were made in the UI or in the API, explain where and if any specific access is needed.

Test you can add the payment methods from the Admin panel and pay using them when creating charges in SGD.

#### 4. Impact of the change

List the steps that must be taken for this PR to work.
E.g.: rake yak:shave, Add "yak_key" to environment variables, ...

Be sure to include all systems that needs to be changed or which system is affected by the change
(Ex: Requires Elastic search to be installed and configured in secrets.yml).

Note: Please provide a screenshot if your changed impact to UI.

New payment methods

#### 5. Priority of change

Normal, High or Immediate.

High

#### 6. Additional Notes

Any further information that you would like to add.

### CODE REVIEW

File: etc/adminhtml/system.xml [hunk 1]
```diff
@@ -363,6 +363,180 @@
...
+                            <field id="allowspecific">1</field>
+                        </depends>
+                    </field>
+                </group>
+                <group id="omise_offsite_kakaopay" translate="label" sortOrder="470" showInDefault="1" showInWebsite="1" showInStore="1">
+                    <label>Enable Kakaopay (Alipay+ Partner) Payment</label>
+                    <comment>Enable customers to pay using Kakaopay payment method</comment>
+                    <field id="active" translate="label comment" type="select" sortOrder="471" showInDefault="1" showInWebsite="1" showInStore="1">
+                        <label>Enable/Disable</label>
+                        <comment>Enable Kakaopay payment method.</comment>
```
Comment (tookaa2) [inline etc/adminhtml/system.xml:487]: a little here, do we want to keep it consistent and use Kakao Pay here also?

**Review by sushil-omise — APPROVED** [review]

File: etc/adminhtml/system.xml (hunk 1 above)
Comment (kan98) [reply etc/adminhtml/system.xml]: good catch! fixed it here https://github.com/omise/omise-magento/pull/309/commits/2a91f6dcb1a397fc8506abc4a9ac7897ab8dcebe

**Review by tookaa2 — APPROVED** [review]

File: Gateway/Request/APMBuilder.php [hunk 2]
```diff
@@ -162,6 +176,42 @@ public function build(array $buildSubject)
...
+                break;
+            case Alipayplus::GCASH_CODE:
+                $paymentInfo[self::SOURCE] = [
+                    self::SOURCE_TYPE   => 'gcash',                
+                    self::PLATFORM_TYPE => $this->helper->getPlatformType(),
+                ];
+                break;
+            case Alipayplus::KAKAOPAY_CODE:
+                $paymentInfo[self::SOURCE] = [
+                    self::SOURCE_TYPE   => 'kakopay',
```
Comment (hpariwat) [inline Gateway/Request/APMBuilder.php:205]: Typo 😄 
```suggestion
                    self::SOURCE_TYPE   => 'kakaopay',                
```

Comment (kan98) [reply Gateway/Request/APMBuilder.php]: thanks 🤦🏽

**Review by Ashraf-alkibsi — APPROVED** [review]

**Review by hpariwat — APPROVED** [review]

**Pull request merged**